src/helpers/configuration_helpers.sh

```diff
@@ -70,6 +103,70 @@ function remove_non_feature_branch {
 }
 
 
+#
+function setup_configuration {
+  setup_configuration_main_branch
+  echo
+  setup_configuration_non_feature_branches
+}
+
+
+function setup_configuration_main_branch {
+  # Ask and store main-branch-name
+  echo "Please enter the name of the main dev branch (typically 'master' or 'development'):"
+  read main_branch_input
+  if [[ -z "$main_branch_input" ]]; then
+    echo_error_header
+    echo_error "You have not provided the name for the main branch."
+    echo_error "This information is necessary to run this script."
+    echo_error "Please try again."
+    exit_with_error
+  fi
+
+  ensure_has_branch "$main_branch_input"
+  store_configuration main-branch-name "$main_branch_input"
+}
+
+
+function setup_configuration_non_feature_branches {
+  # Ask and store non-feature-branch-names
+  echo "Git Town supports non-feature branches like 'release' or 'production'."
+  echo "These branches cannot be shipped and do not merge $main_branch_name when syncing."
+  echo "Please enter the names of all your non-feature branches as a comma separated list."
+  echo "Example: 'qa, production'"
+  read non_feature_input
+
+  if [[ -n "$non_feature_input" ]]; then
+    ensure_has_branches "$non_feature_input" &&
+    ensure_valid_non_feature_branches "$non_feature_input"
```

Comment:
Since this line only executes if the previous one works, you should indent it with 2 spaces to mark it as dependent on the previous one.